component/src/main/java/org/ballerinalang/kafka/util/KafkaUtils.java

```diff
@@ -118,14 +119,29 @@ public static Resource extractKafkaResource(Service service) throws BallerinaCon
 
     private static void validateReturnTypes(BType[] returnParamTypes) {
         for (BType returnParamType : returnParamTypes) {
-            if (returnParamType != BTypes.typeNull && returnParamType != BTypes.typeError) {
```

Comment:
Do we have to iterate the return types here? I mean, we only want to allow `error?` right? For that, can't we simply check whether the length of the return types is 2, and then check whether the type tag of `returnParamTypes[0]` is error, and `returnParamTypes[1]` is nil?